controllers/vspheremachine_controller.go

```diff
@@ -161,7 +161,7 @@ func AddMachineControllerToManager(ctx context.Context, controllerManagerContext
 		).
 		Watches(
 			&clusterv1.Cluster{},
-			handler.EnqueueRequestsFromMapFunc(r.clusterToVSphereMachines),
+			handler.EnqueueRequestsFromMapFunc(r.enqueueClusterToMachineRequests),
```

Comment:
Sorry for stressing this and causing some more work, let's revert the changes around this for this PR and move them to a separate PR.